syncopy/datatype/discrete_data.py

```diff
@@ -617,6 +614,15 @@ def __init__(self,
         :func:`syncopy.definetrial`
 
         """
+        if dimord is not None:
+            # ensure that event data can have extra dimord columns
+            if len(dimord) != len(self._defaultDimord):
+                for col in self._defaultDimord:
+                    if col not in dimord:
+                        base = "dimensional label {}"
+                        lgl = base.format("'" + col + "'")
+                        raise SPYValueError(legal=lgl, varname="dimord")
+                self._defaultDimord = dimord
```

Comment:
Nice! This is a really clever way to append columns to our required `sample` and `eventid` dimord.